### PR TITLE
Remove TiDB-binlog related code

### LINKED ISSUE #341: Remove binlog related pb implement

### PR DESCRIPTION
**Pull request by Benjamin2037**
### What problem does this PR solve?

Issue Number: close #341

Problem Summary:
Remove TiDB-Binlog code, because TiDB-Binlog depricated in v8.3

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### CODE REVIEW

**Review by fzzf678 — DISMISSED** [review]

**Comment by lance6716** [comment]
Also delete the CI "Test go and binlog generation" in `.github/workflows/unit-test.yml`

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -3,23 +3,6 @@ name: "Unit Test"
...
-        with:
-          go-version: "1.20"
-      - name: Install protoc
-        uses: arduino/setup-protoc@v1
-        with:
-          version: '3.5.1'
-      - name: Checkout repository
-        uses: actions/checkout@v2
-      - name: Test
-        run: make go binlog && git diff --exit-code
```
Comment (lance6716) [inline .github/workflows/unit-test.yml:21]: Sorry I see there's `make go` to ensure golang protobuf generation. So this part should be changed to

```
  go:
    name: Test go generation
    runs-on: ubuntu-latest
    steps:
      - name: Set up Go 1.x
        uses: actions/setup-go@v2
        with:
          go-version: "1.20"
      - name: Install protoc
        uses: arduino/setup-protoc@v1
        with:
          version: '3.5.1'
      - name: Checkout repository
        uses: actions/checkout@v2
      - name: Test
        run: make go && git diff --exit-code
```

**Review by lance6716 — APPROVED** [review]

**Pull request merged**